### PR TITLE
Patch #7984 - extended snowplow analytics

### LINKED ISSUE #7984: amp-analytics: extend request parameters of snowplow tracker
Hi, 
this is a feature request for the current implementation for the snowplow tracking sdk, which  is very limited. It would be a helpful improvement if more possible request parameters are added to the request url of the tracker. Especially i miss some basic parameters which provides i.e the page.url, page.referrer or clientId.

- [current implementation amp-analytics](https://github.com/ampproject/amphtml/blob/00e53e2c7b797dbe40898ce20d3e9bb271dfd52f/extensions/amp-analytics/0.1/vendors.js#L960)
- [snowplow tracker protocol](https://github.com/snowplow/snowplow/wiki/snowplow-tracker-protocol)

### PR DESCRIPTION
**Pull request by ecoron**
- extends amp-analytics vendor snowplow by adding more possible properties.

Closes #7984.

### CODE REVIEW

**Comment by ecoron** [comment]
I signed it!

**Comment by ecoron** [comment]
alternative address is signed too

**Comment by ecoron** [comment]
@ampprojectbot retry!

**Comment by jridgewell** [comment]
Hi @ecoron, you'll need to rebase the pull request for our CI tester to pass. See the instructions here: https://github.com/ampproject/amphtml/blob/733281f57b7949e03930e2a4ae0d2d4741ee5b5c/contributing/getting-started-e2e.md#pull-the-latest-changes-from-the-amphtml-repository

**Comment by ecoron** [comment]
i did it, sorry for delay.

File: extensions/amp-analytics/0.1/vendors.js [hunk 1]
```diff
@@ -1123,10 +1123,14 @@ export const ANALYTICS_CONFIG = /** @type {!JSONType} */ ({
...
   'snowplow': {
     'requests': {
-      'aaVersion': 'amp-0.1',
+      'aaVersion': 'amp-0.2',
       'basePrefix': 'https://${collectorHost}/i?url=${canonicalUrl}&page=${title}&' +
           'res=${screenWidth}x${screenHeight}&stm=${timestamp}&' +
-          'tz=${timezone}&aid=${appId}&p=web&tv=${aaVersion}',
+          'tz=${timezone}&aid=${appId}&p=web&tv=${aaVersion}&' +
+          'cd=${screenColorDepth}&cs=${documentCharset}&' +
+          'duid=${clientId}&' +
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: clientId takes a required param as scope (cookie name). this right now throws exception on the analytics-vendors.amp.html example page. @ecoron could you fix that?

/cc @avimehta

Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: Filed an issue for this: #9383

**Pull request merged**